cmd/run.go

```diff
@@ -195,18 +224,38 @@ func newRunCMD(version string) *cobra.Command {
 			}
 
 			g.Go(func() error {
-				logger.Info("wait for policy informer")
+				logger.Info("wait for wgpolicy informer")
 				readinessProbe.Wait()
 
-				logger.Info("start client", zap.Int("worker", c.WorkerCount))
+				logger.Info("start wgpolicy client", zap.Int("worker", c.WorkerCount))
 				for {
+					if wgClient == nil {
```

Comment:
this conditions are on the wrong place, it will continue the loop forever in this case

Reply:
You just Need to move this Check to the top of the function